src/pages/Search.js

```diff
@@ -1,16 +1,35 @@
-import React from 'react';
-import { useParams } from 'react-router-dom';
-import defaultSubreddit from '../helpers/constants';
+import React, { useState, useEffect } from 'react';
+import { useParams, withRouter, useHistory } from 'react-router-dom';
+import SubredditForm from '../components/SubredditForm';
+
 
 const Search = () => {
-  const { subreddit = defaultSubreddit } = useParams();
-  return (
+  const { subreddit: subredditQuery } = useParams();
+  const history = useHistory();
+  const [subreddit, setSubreddit] = useState(subredditQuery);
+
+  const handleSubmit = (evt) => {
+    history.push(`/search/${subreddit}`);
+    evt.preventDefault();
+  };
+
+  const handleChange = (evt) => setSubreddit(evt.target.value);
+
+  useEffect(() => {
+    setSubreddit(subredditQuery);
+  }, [subredditQuery]);
+
+  const searchPage = (
```

Comment:
You could return this directly now that the redirect is gone :smile:

Reply:
Oops! 😅